glean/metrics.yaml

```diff
@@ -0,0 +1,76 @@
+## This file describes the syncserver daily active user (DAU) metrics.
+## This defines the various allowed metrics that are to be captured.
+## Each metric is written as a JSON blob to the default logger output.
+
+---
+# Schema
+$schema: moz://mozilla.org/schemas/glean/metrics/2-0-0
+
+# Category
+syncstorage:
+  sync_event:
+    type: event
+    description: |
+      Event to record an instance of sync backend activity initiated by client.
+    notification_emails:
+      - jrconlin@mozilla.com
+      - pjenvey@mozilla.com
+      - tkorris@mozilla.com
+    bugs:
+      - https://github.com/mozilla-services/syncstorage-rs/issues
+    data_reviews:
+      # TODO
+    expires: never
+
+  sync_user_id:
+    type: uuid
+    # yamllint disable
+    description: >
+      Sync user identifier. Used to determine which user has initiated sync
```

Comment:
"Sync user id" is their numeric id assigned by tokenserver, included in the user's sync endpoint (e.g. `https://sync-1-us-west1-g.sync.services.mozilla.com/1.5/<sync-id>`).

Tokenserver gives users new ids basically whenever their encryption key (represented by `fxa_kid`) changes, such as during password resets without a recovery key.

This makes the sync user id not 100% accurate for counting distinct users: it could count a user multiple times across a time period where their encryption key changed. So a lot like the old fxa query (and I believe the sync client pings) we should use the user's `hashed_fxa_uid` instead.

Reply:
Excellent point regarding the encryption key change. I'll update to reflect the use of `hashed_fxa_uid`.

Reply:
For clarity, I named that metric `hashed_fxa_uid` so there's no confusion on what it is 👍

Reply:
> Any additional concern with deriving said `hashed_fxa_uid` within the context of the application when we'll emit metrics?

Ah yes I was going to mention a btw for when you implement this: tokenserver already sends this value to syncstorage [in its hawk token payload](https://github.com/mozilla-services/syncstorage-rs/blob/1f0e28d7af9c6f9aea38073a099699897464ceac/tokenserver-auth/src/lib.rs#L44). I guess syncstorage has never actually used it so it's not present as a field in its [parsed HawkPayload](https://github.com/mozilla-services/syncstorage-rs/blob/1f0e28d7af9c6f9aea38073a099699897464ceac/syncserver/src/web/auth.rs#L36). You should be able to simply add it as a new field on this struct and then have it available.

I'll note as an FYI the `salt` field in `HawkPayload` is added for us by the Python code (`make_token`) in case you were wondering why it doesn't exist in tokenserver's `MakeTokenPlaintext`.

Then take a look at how we pass around `fxa_uid/kid` via `UserIdentifier`: I believe you'll want to add the hashed version there so it's accessible when emitting the metric.